remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -323,31 +336,40 @@ impl REClientBuilder {
     }
 
     async fn fetch_rbe_capabilities(
-        clients: &mut GRPCClients,
+        client: &mut CapabilitiesClient<GrpcService>,
         instance_name: &InstanceName,
+        max_total_file_size: Option<usize>,
     ) -> anyhow::Result<RECapabilities> {
         // TODO use more of the capabilities of the remote build executor
 
-        let resp = clients
-            .capabilities_client
+        let resp = client
             .get_capabilities(GetCapabilitiesRequest {
                 instance_name: instance_name.as_str().to_owned(),
             })
             .await
             .context("Failed to query capabilities of remote")?
             .into_inner();
-        // Default is a reasonable size for the gRPC transport
-        // with enough room for headers.
-        let mut max_msg_size = DEFAULT_MAX_MSG_SIZE;
+
         let mut exec_enabled = true;
 
-        if let Some(cache_cap) = resp.cache_capabilities {
+        let max_msg_size_from_capabilities: Option<usize> = if let Some(cache_cap) = resp.cache_capabilities {
```

Comment:
Can we rename `max_msg_size` here, but also in the `RECapabilities` type, the buckconfig, etc. to just match/approximately match the `max_batch_total_size_bytes` name that the protocol uses? I feel like that would make this all a bit less confusing to read

Reply:
the 2 are different things actually, which is probably the same type of confusion that created the original bug in the first place :) `max_batch_total_size_bytes` in the protocol does not refer to the max size of the GRPC message and should be strictly lower to that amount, but rather refers to the maximum amount of bytes in the content of BatchFileRead responses. Maybe the difference should be made even clearer?

Reply:
> the 2 are different things actually, which is probably the same type of confusion that created the original bug in the first place

Right, I understand that, that's why my comment. Based on my reading, the value here still has the semantics of the `max_batch_total_size_bytes` value, and it's not until L301 that we use this to get the `max_msg_size` value.

Reply:
yeah, it has the same semantics (intended) and max_msg_size refers to the GRPC limit, but in L301 this meaning doesn't change, I am just ensuring that if no explicit value is provided then max_msg_size is definitely bigger than the max_batch_total_size_bytes or else any batch message of size close to or equal to max_batch_total_size_bytes will always fail to be deserialised. Does it make sense?

Reply:
Right but... the way that I'm reading this code there are two different values:

 1. The max decoding message size, a GRPC property
 2. The max total batch size, the sum of the sizes of the files in a single batch upload request

From my reading, the value returned by the `cache_capabilities`, the value on L355 and the value on L367 all fall into category 2. The value on L282 falls into category 1. L301 then relates the two categories, setting the max decoding message size to twice the max total batch size (unless another value is specified). Is that all correct?

If so, what I'm asking for is that things in category two are all named something like `max_total_batch_size` and things in category 1 all named something like `max_msg_size` or `max_decoding_msg_size`

I don't insist too much though, so would be willing to merge this, as long as you understand what I'm asking

Reply:
ahhhh ok, that makes way more sense now! Thanks :)

Reply:
Hm, maybe we're still miscommunicating, but I would've expected the rename to also include L355, L341, and more importantly, the config property. I won't block on this anymore though, just mentioning it in case you want to change that with the other small nit

Reply:
ah no, I think this was just simply a case of "oooops, missed it", my bad :(

Reply:
anyhow, I fixed this one, thanks for catching it!